eureka-client/src/main/java/com/netflix/appinfo/RefreshableAmazonInfoHolder.java

```diff
@@ -0,0 +1,148 @@
+package com.netflix.appinfo;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A holder class for AmazonInfo that exposes some APIs to allow for refreshes.
+ */
+public class RefreshableAmazonInfoHolder {
```

Comment:
seems like this class should implement standard interface like Supplier or Provider, for better interop with java api